boa/src/exec/mod.rs

```diff
@@ -48,8 +49,12 @@ pub trait Executable {
 /// A Javascript intepreter
 #[derive(Debug)]
 pub struct Interpreter {
-    /// Wether it's running a return statement.
+    /// Whether it's running a return statement.
     is_return: bool,
+
+    /// Whether it's running a break statement.
+    is_break: bool,
```

Comment:
I don't think we can be executing `break` and `return` at the same time. We should probably just have a flag that states `current_state`, which could be `Execution`, `Break`, `Return`, and maybe more in the future, such as `Yield`, `Continue`, `Await`...

This also adds us the option to add a `label` to `break` and `continue` once we implement labeled statements.

Then, we should add a couple of functions `is_return()`, `is_break()`, that would just check this.

Reply:
I agree, would it be best to do this as part of this PR or in a separate one?

Reply:
I think we should do it here. since we only had one flag before, but this adds a second one.

Reply:
Done as part of the most recent commit, the new InterpreterState enum is exposed so that the internal label (or for new states various other information) can be accessed, additionally, I've made it so the label of a break/return node is stored in the interpreter when they run.